### PR TITLE
HWKAPM-449 Include basic zipkin server to intercept span data and con…

### PR DESCRIPTION
**Pull request by objectiser**
…vert for APM use

### CODE REVIEW

**Comment by objectiser** [comment]
@pavolloffay This work is considered experimental at the moment.

File: processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/CommunicationDetailsDeriver.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    @Override
+    public void initialise(String tenantId, List<Span> items) throws RetryAttemptException {
+        producerInfoInitialiser.initialiseFromSpans(tenantId, items);
+    }
+
+    /* (non-Javadoc)
+     * @see org.hawkular.apm.server.api.task.Processor#processSingle(java.lang.Object)
+     */
+    @Override
+    public CommunicationDetails processOneToOne(String tenantId, Span item) throws RetryAttemptException {
```
Comment (pavolloffay) [inline processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/CommunicationDetailsDeriver.java]: `Span item` Could not be named as span?

File: processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/FragmentCompletionTimeDeriver.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+     */
+    public FragmentCompletionTimeDeriver() {
+        super(ProcessorType.OneToOne);
+    }
+
+    /* (non-Javadoc)
+     * @see org.hawkular.apm.server.api.task.Processor#processSingle(java.lang.Object)
+     */
+    @Override
+    public CompletionTime processOneToOne(String tenantId, Span item) throws RetryAttemptException {
```
Comment (pavolloffay) [inline processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/FragmentCompletionTimeDeriver.java]: The same here `Span item`  could not be named as span?

File: processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/CommunicationDetailsDeriver.java (hunk 1 above)
Comment (objectiser) [inline processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/CommunicationDetailsDeriver.java]: It is named after the generic API's parameter (autogenerated by Eclipse). Would prefer to keep the generic name, as consistent across implementations, unless you really think it is a problem?

Comment (pavolloffay) [inline processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/CommunicationDetailsDeriver.java]: I see then, `item` is fine.

**Comment by pavolloffay** [comment]
I went through it, it looks ok, in the next steps we should definitely provide more tests.

**Comment by objectiser** [comment]
@pavolloffay Thanks Pavol! Definitely agree, more tests required.

**Pull request merged**